app/controllers/account_invitations_controller.rb

```diff
@@ -18,8 +21,33 @@ def destroy
     end
   end
 
+  def accept_invitation
+    if params.key?(:invitation_id)
+      redirect_to root_path unless received_invitation && token_valid? && invitation_belongs_to_current_user?
+    else
+      redirect_to root_path
+    end
+  end
+
+  def accept
+    received_invitation.update!(accepted_at: Time.now)
+    redirect_to account_path received_invitation.account_id
+  end
+
   private
 
+  helper_method memoize def received_invitation
+    AccountInvitation.find(ps.fetch(:invitation_id))
+  end
+
+  def invitation_belongs_to_current_user?
+    current_user&.email == received_invitation.email
+  end
+
+  def token_valid?
+    received_invitation.token == ps.fetch(:token)
```

Comment:
In case of review/accept invitation the route should be like this:

`accept_account_invitation/:token`

Then we would not have this weird check for param. These two controllers would have significantly different implementations. Should we create two controllers? The second, for accept, can be named `AcceptAccountInvitationsController` with two standard methods `show` and `update`.